Feature 7

sjw2023 opened the pull request:
[구현]
회원가입 로직
테스트 코드

[학습 내용]
[Unit Test]
[https://docs.google.com/document/d/1YkDdhZQw-l0jTR2J8xP4ox5QyCZNtO_D4HsJHfPoG2Y/edit?usp=sharing](url)
[Exception Handling]
[https://docs.google.com/document/d/1G8eHNoSZkOGyT86QziThQw2TwqmtIFvQNEsRkD8ZHlw/edit?usp=sharing](url)
[Request&Response body]
[https://docs.google.com/document/d/1bblKARyn2UmGUPJ3kq5PgBBc_pgJio0iSYR__cuboTI/edit?usp=sharing](url)
[JSon]
[https://docs.google.com/document/d/17Hie6WYwrTnDox7wBqnFdLvaX0P549ql6__CCN7W3KY/edit?usp=sharing](url)
[Encryption]
[https://docs.google.com/document/d/1jSnVzOIgQQe6EfaYtbzgtT_bCASKkhJsn2gPpVKKgy0/edit?usp=sharing](url)
[Validation]
[https://docs.google.com/document/d/1AvVkBywsrioSGRM9HzGF4ItuZQLm-GFDVemGL19B7Rc/edit?usp=sharing](url)
[RegularExpression]
[https://docs.google.com/document/d/1iWwYTEnN7z_op9PU4H83N_xSa37ORMJ-WNDqFfSfcmo/edit?usp=sharing](url)
[DTO]
[https://docs.google.com/document/d/1hQaV7pX_5CoXf09EAcMfFRXIZMpbSpSOft_hn2rzQrY/edit?usp=sharing](url)

sjw2023 committed:
Adding SignUpDto

sjw2023 committed:
Adding Spring-boot-stater-validation dependency for using validation on DTO.

sjw2023 committed:
add AccountsContoller for handling accounts related request

sjw2023 committed:
add controller to see what happens with Bean Validation Annotation

sjw2023 committed:
change logging level from default to DEBUG to use logger.debug()

sjw2023 committed:
add AccountsService to provide service that are related to Account
sign up is one of Accounts Services

sjw2023 committed:
add AccountsMapper to use MyBatis

sjw2023 committed:
add AccountsMapper.xml to declare the SQL queries for Accounts table

sjw2023 committed:
add INSERT SQL querry to implement sign up service

sjw2023 committed:
add member field accountsMapper and inject mapper into service
implements signUp() using accountsMapper

sjw2023 committed:
add AccountsService as member field to inject
implement signUp() using accountsService

sjw2023 committed:
change mapper name space it should point exact location of mapper

sjw2023 committed:
change table name in SQL query since talble named in lowercase

sjw2023 committed:
feature-7 sign up service implementation with test codes.

sjw2023 committed:
erase test code files

sjw2023 committed:
clean up the comments

sjw2023 committed:
erase remain debug print

sjw2023 committed:
change all the field name to camel case

sjw2023 committed:
moved exception handler to gather all the exception handler in to one package

rename class ExceptionHandler since it is conflicting with @ExceptionHandler annotation

sjw2023 committed:
replace String to StringBuilder for better performance and thread-safe

sjw2023 committed:
add controller advice explanation

sjw2023 committed:
internationalization test

sjw2023 committed:
fix typo

sjw2023 committed:
fix typy after changing field name in DTO

sjw2023 committed:
add AccountsDao class

sjw2023 committed:
change signUp to use dao

sjw2023 committed:
change insertAccount to use AccountsDao

sjw2023 committed:
add isIdExist query in AccountMapper

sjw2023 committed:
add isIdExist quary

sjw2023 committed:
change isIdExist to reaturn boolean

sjw2023 committed:
fix syntax error in sql

sjw2023 committed:
add IdExistException and hanlder

sjw2023 committed:
change to use ResourceBundleMessageSource

sjw2023 committed:
change exception to have errorCode

sjw2023 committed:
change exception handling for more general situation

sjw2023 committed:
add insert failure case and throwing if insert fails

sjw2023 commented:
[구현 내용]
ErrorCode구현
회원가입, 중복체크시 예외처리 추가
[추가 학습 내용}
[Error Code & Enum & ResourceBundle]
https://docs.google.com/document/d/15LPSp41JugrA62t546-iRO51Gl2QaQms-MW1b7Jqu98/edit?usp=sharing
[Transactional]
https://docs.google.com/document/d/12cMa_r2mL3GOBlF93lWMd3YNplk7n___wlI-OD7q2-g/edit?usp=sharing
[Repository 와 Mapper 그 차이점]
https://docs.google.com/document/d/1iyOgjHh5PXjTZVDAAATJm6Ww7-2S296ha96JJxv650s/edit?usp=sharing

f-lab-robert commented on build.gradle:
```diff
@@ -15,9 +15,10 @@ repositories {
 	implementation 'org.springframework.boot:spring-boot-starter-web'
 	implementation 'org.mybatis.spring.boot:mybatis-spring-boot-starter:2.2.2'
+	implementation 'org.springframework.boot:spring-boot-starter-validation:2.7.5'
+    implementation 'org.jetbrains:annotations:20.1.0'
```
정렬이 안되어있네요.

sjw2023 replied:
![정렬](https://user-images.githubusercontent.com/16884433/204965241-e432b476-5042-4feb-bd2e-9ad3753e44ad.png)
음 제 쪽에는 정렬 되어있는것으로 출력됩니다

f-lab-robert commented on src/main/java/com/example/myinsta/controller/AccountsController.java:
```diff
@@ -0,0 +1,50 @@
+    public ResponseEntity<Void> signUp(@Valid @RequestBody SignUpDto signUpDto){
+        accountsService.signUp( signUpDto );
+        return new ResponseEntity(HttpStatus.CREATED);
+    }
```
1. 포멧팅에 대해 신경써주시면 좋을 것 같습니다. 우선 IntelliJ 단축키를 통해 코드 정렬하는 방법에 대해 살펴봐주세요.
2. private final AccountsService accountsService; 위 아래로 1줄씩 공백을 만들어주시고
3. accountsService.signUp( signUpDto ); -> () 괄호 안에 불필요한 공백은 없애주세요

sjw2023 replied:
알겠습니다. 앞으로 메소드 () 안에 공백 없애고 맴버 필드 선언시 위 아래 한줄씩 공백 추가 하도록 하겠습니다. option+cmd+L키로 정렬 하는법 사용 하였습니다. 기본 포맷말고 추천해주실 포맷 알려주시면 적용 시키 도록 하겠습니다.

f-lab-robert commented on src/main/java/com/example/myinsta/dao/AccountsDao.java:
```diff
@@ -0,0 +1,21 @@
+@Getter
+@Setter
+@Builder
+@ToString
```
Data annotation 을 활용하는 할 수는 없을까요 ?

sjw2023 replied:
가능 합니다 바꾸도록 하겠습니다

f-lab-robert commented on src/main/java/com/example/myinsta/dao/AccountsDao.java:
```diff
@@ -0,0 +1,21 @@
+    private String nickName;
+    private String password;
+    private Timestamp createdDate;
+    private Timestamp updatedDate;
```
LocalDate, LocalDateTime 에 대해 찾아보시겠어요 ?

sjw2023 replied:
확인하였습니다, LocalDate는 MySQL TimeStamp의 YYYY-MM-DD를 표현하며 LocalDateTime은 내부에 LocalDate + LocalTime을 가지고 YYYY-DD-MM HH-MM-SS로 표현된다는걸 확인하엿으며, java.sql.TimeStamp.toLocalDateTime()을 통해 변환이 가능한점 확인했습니다, 자바 타입인 LocalDateTime으로 바꾸도록 하겠습니다.

f-lab-robert commented on src/main/java/com/example/myinsta/exception/AccountsExceptionHandler.java:
```diff
@@ -0,0 +1,63 @@
+    ResourceBundleMessageSource messageSource;
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<Object> handler(MethodArgumentNotValidException e){
```
1. MethodArgumentNotValidException e 의 `e` 를 통해 에러베세지를 추출할 수는 없을까요 ? 
2. 더불어 제가 다른 멘티분들의 코드를 공유드리면서 Enum 형태로 정리하면 좋을 것 같다고 했었는데요. 기존 방식을 그대로 고수하시는 이유가 있을까요 ? 즉, 아래의 코드는 다른 형태로 작성하셨는데, 아래 포맷을 지켜보는 방식은 어떨까요 ?

sjw2023 replied:
1. 가능 합니다 
2. 죄송합니다. 윗 부분도 같이 바꿧어야 했는데 아래만 생각하고 잊었습니다. 바꿔놓도록 하겠습니다.

f-lab-robert commented on src/main/java/com/example/myinsta/exception/AccountsExceptionHandler.java:
```diff
@@ -0,0 +1,63 @@
+@Slf4j
+public class AccountsExceptionHandler {
+    @Autowired
+    ResourceBundleMessageSource messageSource;
```
아래의 코멘트를 참고해봐주세요. 참고해보시면, 이 부분은 필요가 없을 것 같습니다.

f-lab-robert commented on src/main/java/com/example/myinsta/exception/AccountsExceptionHandler.java:
```diff
@@ -0,0 +1,63 @@
+ *
+ */
+@ControllerAdvice
+@Slf4j
```
사용하지 않는 어노테이션은 지워주세요

sjw2023 replied:
넵

f-lab-robert commented on src/main/java/com/example/myinsta/exception/AccountsExceptionHandler.java:
```diff
@@ -0,0 +1,63 @@
+        json.append("}");
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST).contentType(MediaType.APPLICATION_JSON).body(json);
+    }
+    @ExceptionHandler({IdExistException.class, InsertFailException.class})
```
IdExistException -> 사용자의 잘못 400
InsertFailException -> 예외 처리를 못한 서버의 잘못 500

으로 구분되어야 하지 않을까요

sjw2023 replied:
넵

f-lab-robert commented on src/main/java/com/example/myinsta/mapper/AccountsMapper.java:
```diff
@@ -0,0 +1,12 @@
+import org.apache.ibatis.annotations.Mapper;
+
+//@Mapper : To be scanned by mapper-scanner of mybatis.
+@Mapper
```
@Repository 기억하시나요 ?

sjw2023 replied:
넵, [스프링 공식 문서](https://docs.spring.io/spring-data/data-commons/docs/1.6.1.RELEASE/reference/html/repositories.html)에 의하면 @Repository는 JPA같은 ORM프레임워크를 사용할때 사용하는 annotation으로 @Repository로 마크된 interface는 스프링에서 기본 제공되는 JPARepository등의 인터페이스를 상속하여 사용합니다. 상속시 사용할 DAO(Entity, VO)를 같이 선언하여 멤버 필드를 상속 받을수 있으며 스프링에서 정의한 키워드를 사용하여 멤버 method를 정의해두면 스프링이 알아서 ORM을 사용하여 SQL 질의문을 생성해주도록 하는 어노테이션인것을 확인 하였습니다.

이에 반해 @Mapper는 단순히 사용자가 생성한 SQL 질의문을 스프링에서 사용할수있게 interface내부의 메소드와 매핑 시켜주는 어노테이션입니다. 

이것으로 보아 스프링은  @Repository와 ORM들로 MyBatis의 역할을 추상화 해주고 있는것으로 보입니다. PSA라고 생각했던 입맛에 맞는 ORM을 바꿔 끼우기 편한 구조는 추상화를 한 결과의 부가적인것으로 보입니다.

f-lab-robert commented on src/main/java/com/example/myinsta/exception/IdExistException.java:
```diff
@@ -0,0 +1,11 @@
+
+@Getter
+@RequiredArgsConstructor
+public class IdExistException extends RuntimeException{
```
앞으로 Exception 의 종류를 이렇게 여러개 선언하실건가요 ?
즉, A Exception, B Exception .... Z Excetion... 너무 많아질 것 같지 않나요 ?

- https://github.com/f-lab-edu/ec_mall/blob/main/src/main/java/com/example/ec_mall/exception/APIException.java 

위의 예시처럼 하나의 클래스를 선언하고 에러 메세지를 통해 구분할 수 있지 않을까요 ?

sjw2023 replied:
넵 General Exception 클래스를 생성하여 ErrorCode를 통해 구분하는 방식으로 바꾸도록 하겠습니다.

f-lab-robert commented on src/main/resources/messages.properties:
```diff
@@ -0,0 +1,3 @@
+700=The sign-up information is not valid
+701=Email already exists
+702=Sign-up failed
```
위의 부분들은 저희의 로직입니다. 즉 코드 안으로 들어와야겠죠 ?
- https://github.com/f-lab-edu/ec_mall/blob/main/src/main/java/com/example/ec_mall/exception/ErrorCode.java

위의 예제를 참고해서 고쳐보시겠어요 ?

sjw2023 replied:
코드 안으로 Enum을 이용하여 넣어 주었습니다.

sjw2023 committed:
formatting texts

sjw2023 committed:
change annotation on DAO

sjw2023 committed:
change handler to use ErrorResponse

sjw2023 committed:
change from individual exception to general purpose exception
change ErrorCode to use enum type

sjw2023 committed:
remove not using annotation

f-lab-robert commented on src/main/java/com/example/myinsta/dao/AccountsDao.java:
```diff
@@ -0,0 +1,23 @@
+public class AccountsDao {
+
+    private long idAccount;
+
```
불필요한 개행은 지양해주세요. 여러 필드들을 굳이 개행할 필요는 없을 것 같습니다.

sjw2023 replied:
넵

f-lab-robert commented on src/main/java/com/example/myinsta/exception/AccountsExceptionHandler.java:
```diff
@@ -0,0 +1,54 @@
+        List<FieldError> errors = e.getFieldErrors();
+        List<ErrorResponse> errorResponse = new ArrayList<>();
+        for (FieldError error : errors) {
+            errorResponse.add(ErrorResponse.builder().errorCode(700).errorMessage(error.getRejectedValue() + ", " + error.getDefaultMessage()).build());
```
700 상태는 Enum 으로 관리하고 있는 부분에서 가져와서 사용해야 하지 않을까요 ?

sjw2023 replied:
넵

f-lab-robert commented on src/main/java/com/example/myinsta/exception/AccountsException.java:
```diff
@@ -0,0 +1,10 @@
+
+@Getter
+@RequiredArgsConstructor
+public class AccountsException extends RuntimeException{
```
General 하게 사용하고 있지 않은 것 같은데요 ?
CustomException, GeneralException, APIException 등의 형태가 되어야 하지 않나요 ?

sjw2023 replied:
넵

f-lab-robert commented on src/main/java/com/example/myinsta/controller/AccountsController.java:
```diff
@@ -0,0 +1,63 @@
+    @PostMapping("/accounts/signup")
+    public ResponseEntity<Void> signUp(@Valid @RequestBody SignUpDto signUpDto) {
+        accountsService.signUp(signUpDto);
+        return new ResponseEntity(HttpStatus.CREATED);
```
IntelliJ 에 경고창으로 Generic 을 활용하라고 나오지 않나요 ?

sjw2023 replied:
넵

f-lab-robert commented on src/main/java/com/example/myinsta/service/AccountsService.java:
```diff
@@ -0,0 +1,60 @@
+                .nickName(signUpDto.getNickName())
+                .password(SHA256.encrypt(signUpDto.getPassword()))
+                .build();
+        if (accountsMapper.isIdExist(accountsDao) == true) {
```
IntelliJ 에 == true 는 필요 없다고 안내해주지 않나요 ?

sjw2023 replied:
넵

f-lab-robert commented on src/main/java/com/example/myinsta/service/AccountsService.java:
```diff
@@ -0,0 +1,60 @@
+        } else {
+            if (accountsMapper.insertAccount(accountsDao) != 1) {
+                throw new AccountsException(ErrorCode.FAILED_TO_INSERT);
+            }
```
- 로직은 논리 단위가 명확하게 구분이 되어야 합니다. 현재는 if 문을 제대로 활용하는 방안에 대한 고찰이 부족해보입니다.
```
if(accountsMapper.isIdExist(accountsDao)){
   throw new AccountsException(ErrorCode.ALREADY_EXIST_EMAIL);
} 

int result = accountsMapper.insertAccount(accountsDao);
if(result !=1 ){
  throw new AccountsException(ErrorCode.FAILED_TO_INSERT);
} 
```

현 코드 기준으로는 위와 같이 구분 짓는게 로직의 단위가 명확해지겠죠 ? 그리고 else 가 필요하지 않은데, 불필요하게 현재 들어가 있는 부분도 제거가 되고요

sjw2023 replied:
넵 명심하겠습니다

f-lab-robert commented on src/test/java/com/example/myinsta/AccountsControllerTest.java:
```diff
@@ -0,0 +1,96 @@
+    @MockBean
+    AccountsService accountsService;
+
+    @Test
```
1. 무엇을 테스트하고 싶으신 건지 제 3자는 어떻게 알 수 있을까요 ?
2. DisplayName 을 통해서 테스트하고 싶은 경우를 명확하게 적어주시고. 메소드 명 또한 그에 맞게 수정해주세요. 추상적인게 아닌 구체적인 시나리오를 정해서 테스트하셔야 합니다.
3. Invalid 한 경우가 과연 1개만 있을까요 ? 간단하게만 생각해봐도 최소 3개 이상은 떠오르는 것 같은데요.

sjw2023 replied:
넵

f-lab-robert commented on src/test/java/com/example/myinsta/AccountsControllerTest.java:
```diff
@@ -0,0 +1,96 @@
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(objectMapper.writeValueAsString(signUpDto)))
+                .andDo(print())
+                .andExpect(status().isBadRequest());
```
실제로 사용자가 잘못 입력했을 때, 주원님이 만들어 놓이신 예외 메세지가 나오는지를 확인해봐야 하지 않을까요 ? 단순 400 에러만 체크를 하면 실제 사용자에게 내가 원하는 메세지가 전달되었는지는 어떻게 보장하시나요 ?

sjw2023 replied:
컨텐츠를 스트링으로 비교하도록 해두었습니다.

f-lab-robert commented on src/test/java/com/example/myinsta/AccountsControllerTest.java:
```diff
@@ -0,0 +1,96 @@
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @MockBean
```
Mock, MockBean 등 아무런 이해가 현재는 없어보입니다. 테스트 강좌를 듣고 다시와서 봐주세요.

sjw2023 replied:
넵

f-lab-robert commented on src/test/java/com/example/myinsta/AccountsServiceTest.java:
```diff
@@ -0,0 +1,48 @@
+
+@Slf4j
+@ExtendWith(MockitoExtension.class)
+public class AccountsServiceTest {
```
이 부분에 관한 부분은 금주 멘토링 시간에 같이 이야기를 나누어볼게요.

sjw2023 committed:
remove syntax warning

sjw2023 committed:
annotate test method with @DisplayName and change method names

sjw2023 committed:
add more invalid sign-up information test

sjw2023 committed:
add test if result is correct

sjw2023 committed:
put more detail on @WebMvcTest explanation

sjw2023 committed:
put more detail on @MockBean explanation

sjw2023 committed:
remove test from test methods name

sjw2023 committed:
put more detail on DisplayName

sjw2023 committed:
change exception name and handler name for understanding

sjw2023 committed:
add stubbing on mock object

sjw2023 committed:
change to use JSonPath when test

sjw2023 committed:
remove wrong value from error response message for security purpose
chage error message to start with upper-case letter

sjw2023 committed:
change validation annotation to remove overlapped EmptyString check

sjw2023 committed:
added new line for praticing purpose

sjw2023 committed:
change MethodArgumentNotValidException Handler to return one invalid field when validation fails more than 1 times
3 data invalid case :
	return Email when 3 data is incorrect
2 data invalid case :
	return Email when Email and (nickName|password) are incorrect
	return nickName when nickName and password are incorrect

sjw2023 committed:
setup given state for service test
add 3 tests for chekcing exception test-case

sjw2023 committed:
add successful situation of sign-up\n not throwing any exceptions and all queries exceuted atleast once

sjw2023 committed:
remove Boilerplate with @BeforeEach

sjw2023 committed:
add explanation

sjw2023 committed:
change to use @Mock and @InjectMock

f-lab-robert commented on src/main/java/com/example/myinsta/controller/AccountsController.java:
```diff
@@ -0,0 +1,62 @@
+ * @param signUpDto
+ * @return ResponseEntity with status code CREATED when INSERT success
+ * @return Error message when data validation fails or INSERT query fails
+ */
```
위의 부분들은 차후에 docs or Wiki 로 따로 빼서 정리하는게 어떨까 싶습니다.

f-lab-robert commented on src/main/java/com/example/myinsta/exception/AccountsExceptionHandler.java:
```diff
@@ -0,0 +1,75 @@
+    public ResponseEntity<Object> handler(MethodArgumentNotValidException e) {
+        List<FieldError> errors = e.getFieldErrors();
+        ErrorResponse errorResponse = null;
+        for (FieldError error : errors){
```
너무 난잡하고 일반화되어 있지 않습니다. if else 가 이런식으로 들어가는 것은 바람직하지 못합니다. 즉, 일반화되서 사용되는게 아니라 특정 로직에 종속적인 형태가 되어가고 있습니다. error 중에 제일 첫번째 것을 가져와서 사용자에게 보여주는 형태로 고치는게 좋을 것 같습니다.

f-lab-robert commented on src/test/java/com/example/myinsta/AccountsServiceTest.java:
```diff
@@ -0,0 +1,113 @@
+    void sign_up_throw_exception_when_id_already_exist() {
+        //given
+        given(accountsMapper.isIdExist(any(AccountsDao.class))).willReturn(true);
+        willDoNothing().given(accountsMapper.insertAccount(any(AccountsService.class)));
```
IntelliJ 에서 이 부분에 오류가 있다고 나오는데 실제로 테스트를 돌려보신게 맞을까요 ?

f-lab-robert commented on src/test/java/com/example/myinsta/AccountsControllerTest.java:
```diff
@@ -0,0 +1,203 @@
+                .andExpect(jsonPath(errorMessage).value("Invalid Email format"))
+//                .andExpect(jsonPath("$[1].errorMessage").value("NickName must not null or empty string"))
+//                .andExpect(jsonPath("$[1].errorMessage").value("NickName should be in range of 1 to 16"))
+//                .andExpect(jsonPath("$[2].errorMessage").value("Password must have at least 8 characters with maximum 16 characters, one Upper case, one number, one symbol."))
```
주석 처리 되어 있는 내용들은 무엇일까요

f-lab-robert commented on src/test/java/com/example/myinsta/AccountsControllerTest.java:
```diff
@@ -0,0 +1,203 @@
+
+
+
+
```
불필요한 공백들이 너무 많이 보입니다. 신경써주세요

f-lab-robert commented on src/main/java/com/example/myinsta/utill/SHA256.java:
```diff
@@ -0,0 +1,31 @@
+ * Encryption class that uses java.security SHA256
+ */
+public class SHA256 {
+    public static String encrypt(String originalString) {
```
이 로직에 대한 TestCode 도 필요하지 않을까요 ? encrypt 가 제대로 되었는지 어떻게 검증하실 수 있을까요

f-lab-robert commented on src/test/java/com/example/myinsta/AccountsServiceTest.java:
```diff
@@ -0,0 +1,113 @@
+        given(accountsMapper.isIdExist(any(AccountsDao.class))).willReturn(false);
+        given(accountsMapper.insertAccount(any(AccountsDao.class))).willReturn(0);
+        //when
+        assertThrows(CustomException.class, () -> accountsService.signUp(signUpDto));
```
저희는 service 를 테스트하고 있습니다. 즉 서비스가 이미 에러가 발생한다라고 가정을 해버리는 것은 controller 에서 가정해야될 상황이지 service 내부 로직을 테스트하는 현재 단계에서는 가정이 잘못 된 것 같습니다. 즉, service 내부의 특정 mapper 에서 에러가 생겼을 때를 가정해야 하지 않을까요 ?

f-lab-robert commented on src/test/java/com/example/myinsta/AccountsServiceTest.java:
```diff
@@ -0,0 +1,113 @@
+        given(accountsMapper.isIdExist(any(AccountsDao.class))).willReturn(false);
+        given(accountsMapper.insertAccount(any(AccountsDao.class))).willReturn(1);
+        //when
+        assertDoesNotThrow(() -> accountsService.signUp(signUpDto));
```
위와 동일합니다.

f-lab-robert commented on src/test/java/com/example/myinsta/AccountsServiceTest.java:
```diff
@@ -0,0 +1,113 @@
+        given(accountsMapper.isIdExist(any(AccountsDao.class))).willReturn(false);
+        given(accountsMapper.insertAccount(any(AccountsDao.class))).willReturn(1);
+        //when
+        assertDoesNotThrow(() -> accountsService.signUp(signUpDto));
```
위와 동일합니다.

sjw2023 committed:
remove comment

sjw2023 committed:
change exception handler to return first value

sjw2023 committed:
replace boilerplate with @BeaforeEach

sjw2023 committed:
remove multiple invalid input test

sjw2023 committed:
remove boilerplate with @BeforeEach

sjw2023 committed:
change explanation in @DisplayName for better understanding

sjw2023 committed:
feature-7: change explanation in @DisplayName for better understanding

sjw2023 committed:
feature-7: given for each test case should different

sjw2023 committed:
feature-7: change test code to check message of actual thrown exception

sjw2023 committed:
feature-7: fix typo in test method name

Commit:
feature-7: remove meaningless assert and incorrect capture() ussing
